input_test_cases/run_all_tests.py

```diff
@@ -0,0 +1,217 @@
+
+    Parameters
+    ----------
+    input_dir : str
+        Directory where the input files are located.
+    input_json : str
+        JSON file with input parameters.
+    output_dir : str
+        Directory where the output files are generated.
+    desired_dir : str
+        Directory where the desired output files are located.
+    """
+    os.chdir(input_dir)
+    subprocess.run(["ossdbs", input_json], check=True)
+    os.chdir(output_dir)
+    actual = pd.read_csv("impedance.csv").to_numpy()
+    os.chdir(desired_dir)
+    desired = pd.read_csv("impedance.csv").to_numpy()
+    os.chdir("../../..")
+    np.testing.assert_allclose(actual, desired, atol=1e-5)
+
+
+def test_VTA(output_dir: str, output_nii: str, desired_dir: str, desired_nii: str):
+    """
+    Tests VTA by comparing the actual output with the desired output
+    using the Dice coefficient.
+
+    Parameters
+    ----------
+    output_dir : str
+        Directory where the output files are located.
+    output_nii : str
+        NIfTI file with the output data.
+    desired_dir : str
+        Directory where the desired output files are located.
+    desired_nii : str
+        NIfTI file with the desired data.
+    """
+    os.chdir(output_dir)
+    actual = _readNIfTI(output_nii)
+    os.chdir(desired_dir)
+    desired = _readNIfTI(desired_nii)
+    os.chdir("../../..")
+    assert _compute_dice_coefficient(actual, desired) == 1
+
+
+def _readNIfTI(filename: str) -> set:
+    """
+    Reads a NIfTI file.
+
+    Parameters
+    ----------
+    filename : str
+        Path to the NIfTI file.
+    """
+    image = nib.load(filename)
+    affine = image.affine
+    data_shape = image.get_fdata().shape
+    data = image.get_fdata()
+    vta_points = []
+
+    end_voxel = [data_shape[0], data_shape[1], data_shape[2]]
+    counter = 0
+    for i in range(0, end_voxel[0]):
+        for j in range(0, end_voxel[1]):
+            for k in range(0, end_voxel[2]):
+                if data[i, j, k] > 0.1:
+                    counter += 1
+                    X_coord = affine[0][3] + i * affine[0][0]
+                    Y_coord = affine[1][3] + j * affine[1][1]
+                    Z_coord = affine[2][3] + k * affine[2][2]
+                    vta_points.append((X_coord, Y_coord, Z_coord))
+
+    return set(vta_points)
+
+
+def _compute_dice_coefficient(set1, set2) -> float:
+    """
+    Computes the Dice coefficient.
+
+    Parameters
+    ----------
+    set1 : set
+        First set of points.
+    set2 : set
+        Second set of points.
+    """
+    intersection = set1.intersection(set2)
+    intersection_size = len(intersection)
+    size1 = len(set1)
+    size2 = len(set2)
+    return (2 * intersection_size) / (size1 + size2)
+
+
+# input_case1
+test_impedance(
+    input_dir="input_case1",
+    input_json="input_homogeneous.json",
+    output_dir="Results_homogeneous",
+    desired_dir="../../desired_output/input_case1/Results_homogeneous",
```

Comment:
These directories are missing, can you please add them? Ideally, they are all in `input_test_cases`